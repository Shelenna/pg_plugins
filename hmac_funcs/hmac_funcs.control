# hmac_funcs extension
comment = 'HMAC functions'
default_version = '1.0'
module_pathname = '$libdir/hmac_funcs'
relocatable = true
