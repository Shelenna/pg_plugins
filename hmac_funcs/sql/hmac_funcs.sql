CREATE EXTENSION hmac_funcs;
SELECT hmac_md5('The quick brown fox jumps over the lazy dog', 'key');
SELECT hmac_sha1('The quick brown fox jumps over the lazy dog', 'key');
SELECT hmac_sha224('The quick brown fox jumps over the lazy dog', 'key');
SELECT hmac_sha256('The quick brown fox jumps over the lazy dog', 'key');
SELECT hmac_sha384('The quick brown fox jumps over the lazy dog', 'key');
SELECT hmac_sha512('The quick brown fox jumps over the lazy dog', 'key');
