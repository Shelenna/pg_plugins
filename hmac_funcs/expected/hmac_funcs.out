CREATE EXTENSION hmac_funcs;
SELECT hmac_md5('The quick brown fox jumps over the lazy dog', 'key');
              hmac_md5              
------------------------------------
 \x80070713463e7749b90c2dc24911e275
(1 row)

SELECT hmac_sha1('The quick brown fox jumps over the lazy dog', 'key');
                 hmac_sha1                  
--------------------------------------------
 \xde7c9b85b8b78aa6bc8a7a36f70a90701c9db4d9
(1 row)

SELECT hmac_sha224('The quick brown fox jumps over the lazy dog', 'key');
                        hmac_sha224                         
------------------------------------------------------------
 \x88ff8b54675d39b8f72322e65ff945c52d96379988ada25639747e69
(1 row)

SELECT hmac_sha256('The quick brown fox jumps over the lazy dog', 'key');
                            hmac_sha256                             
--------------------------------------------------------------------
 \xf7bc83f430538424b13298e6aa6fb143ef4d59a14946175997479dbc2d1a3cd8
(1 row)

SELECT hmac_sha384('The quick brown fox jumps over the lazy dog', 'key');
                                            hmac_sha384                                             
----------------------------------------------------------------------------------------------------
 \xd7f4727e2c0b39ae0f1e40cc96f60242d5b7801841cea6fc592c5d3e1ae50700582a96cf35e1e554995fe4e03381c237
(1 row)

SELECT hmac_sha512('The quick brown fox jumps over the lazy dog', 'key');
                                                            hmac_sha512                                                             
------------------------------------------------------------------------------------------------------------------------------------
 \xb42af09057bac1e2d41708e48a902e09b5ff7f12ab428a4fe86653c73dd248fb82f948a549f7b791a5b41915ee4d1ec3935357e4e2317250d0372afa2ebeeb3a
(1 row)

